lib/components/primitive-components/Keepout.ts

```diff
@@ -0,0 +1,56 @@
+import { PrimitiveComponent } from "../base-components/PrimitiveComponent"
+import { pcbKeepoutProps } from "@tscircuit/props"
+import type { RenderPhaseFn } from "../base-components/Renderable"
+import type { PCBKeepout } from "@tscircuit/soup"
+import { decomposeTSR } from "transformation-matrix"
+
+export class KeepOut extends PrimitiveComponent<typeof pcbKeepoutProps> {
+  pcb_keepout_id: string | null = null
+
+  isPcbPrimitive = true
+
+  get config() {
+    return {
+      zodProps: pcbKeepoutProps,
+    }
+  }
+
+  doInitialPcbPrimitiveRender(): void {
+    const { db } = this.root!
+    const { _parsedProps: props } = this
+    const position = this._getGlobalPcbPositionBeforeLayout()
+    const decomposedMat = decomposeTSR(
+      this._computePcbGlobalTransformBeforeLayout(),
+    )
+    const isRotated90 =
+      Math.abs(decomposedMat.rotation.angle * (180 / Math.PI) - 90) < 0.01
+
+    let pcb_keepout: PCBKeepout | null = null
+    if (props.shape === "circle") {
+      pcb_keepout = db.pcb_keepout.insert({
+        layer: ["top"],
+        shape: "circle",
+        // @ts-ignore: no idea why this is triggering
+        radius: props.radius,
+        center: {
+          x: position.x,
+          y: position.y,
+        },
```

Comment:
the rect vs circle should not have their centers defined differently, we should fix this in soup before encoding it into core

Reply:
https://github.com/tscircuit/circuit-json/issues/40